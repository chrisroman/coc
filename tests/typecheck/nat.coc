let nat = [C:*][_: C][_: [_: C]C]C in
let zero = (\C:*) (\z:C) (\f: [_: C]C) z in
let zero = (\C:*) (\z:C) (\f: [_: C]C) z in
let idT = [t: *][x: t]t in
let id = (\t: *)(\x: t)x in
let fn = (id idT) in
(((zero idT) id) fn)
