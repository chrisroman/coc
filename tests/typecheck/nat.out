let nat = [C: *] [_: C] [_: [_: C] C] C in 
let zero = (λC: *) (λz: C) (λf: [_: C] C) z in 
let zero = (λC: *) (λz: C) (λf: [_: C] C) z in 
let idT = [t: *] [x: t] t in 
let id = (λt: *) (λx: t) x in 
let fn = (id idT) in 
(((zero idT) id) fn)
{nat} typecheck_single_term [C: *] [_: C] [_: [_: C] C] C
Type: *
{zero} typecheck_single_term (λC: *) (λz: C) (λf: [_: C] C) z
Type: [C: *] [z: C] [f: [_: C] C] C
{zero} typecheck_single_term (λC: *) (λz: C) (λf: [_: C] C) z
Type: [C: *] [z: C] [f: [_: C] C] C
{idT} typecheck_single_term [t: *] [x: t] t
Type: *
{id} typecheck_single_term (λt: *) (λx: t) x
Type: [t: *] [x: t] t
{fn} typecheck_single_term ((λt: *) (λx: t) x [t: *] [x: t] t)
Type: [x: [t: *] [x: t] t] [t: *] [x: t] t
typecheck_single_term ((((λC: *) (λz: C) (λf: [_: C] C) z [t: *] [x: t] t) (λt: *) (λx: t) x) (λx: [t: *] [x: t] t) x)
Type: [t: *] [x: t] t
Value: (λt: *) (λx: t) x
